kube-runtime/src/controller/mod.rs

```diff
@@ -506,7 +528,21 @@ where
         let reader = writer.as_reader();
         let mut trigger_selector = stream::SelectAll::new();
         let self_watcher = trigger_self(
-            reflector(writer, watcher(owned_api, lp)).applied_objects(),
+            reflector(
+                writer,
+                watcher(owned_api, lp).map_ok(|mut ev| {
+                    let strip_mf = |o: &mut K| {
+                        o.meta_mut().managed_fields = None;
+                    };
```

Comment:
I'm still not sure if we should do this by default. What if controllers are using this in their reconcilers?

Reply:
Agreed. This is also largely a lot less relevant when `Event::modify` exists. Not sure how I missed that before.

Reply:
An alternative would be to add a watcher config parameter for it... but the cost/reward isn't really there when we have `Controller::for_stream` now.

Reply:
Removed in 8078648ab1e25fbce126a9440b988559281a7565